src/soap/handlers/GetSiteInfoObject.js

```diff
@@ -54,7 +54,7 @@ export async function* getSiteInfoObject(
   { date = new Date(), helpers, method, parameters, uniqueid }
 ) {
   const { site } = parameters
-  const parts = site.split(':')
+  const parts = site && site.split(':')
```

Comment:
I was playing the the original SOAP service and it's just a plain headache. It appears to return an error when the site is not valid. Suggestion is to change this to `const parts = (site && site.split(':')) || []`.

Then below the slug is not optional. See next comment.